examples/custom_solvers.jl

```diff
@@ -0,0 +1,47 @@
+# # Custom solvers
+# In this example, we show how to define custom solvers. Our system will again be silicon, because we are not very imaginative
+using DFTK, LinearAlgebra
+
+a = 10.26
+lattice = a / 2 * [[0 1 1.];
+                   [1 0 1.];
+                   [1 1 0.]]
+Si = ElementPsp(:Si, psp=load_psp("hgh/lda/Si-q4"))
+atoms = [Si => [ones(3)/8, -ones(3)/8]]
+
+## We take very (very) crude parameters
+model = model_LDA(lattice, atoms)
+kgrid = [1, 1, 1]
+Ecut = 5
+basis = PlaneWaveBasis(model, Ecut; kgrid=kgrid);
+
+# We define our custom fix-point solver: simply a damped fixed-point
+function my_fp_solver(f, x0, max_iter; tol)
+    mixing_factor = .7
+    x = x0
+    fx = f(x)
+    for n = 1:max_iter
+        inc = fx - x
+        ## Note that the convergence criterion might be overwritten by the SCF solver
```

Comment:
This needs a bit more explanation. Also I would not have it here, but discuss it in the pretext.